### PR TITLE
Add new Error Boundary for Gallery Page and handle 404 correctly

### PR DESCRIPTION
**Pull request by kaitoo1**
Changes:
- Added new Error Boundary for Gallery Page to specifically handle 404 case
- Updated ApiError class to include response code
- 404 view is now rendered without redirecting to /404 route
- Updated 404 view message

### CODE REVIEW

File: src/scenes/NotFound/NotFound.tsx [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 import Page from 'components/core/Page/Page';
 
-function NotFound(_: RouteComponentProps) {
+function NotFound() {
   return (
     <Page centered>
       <Display>404</Display>
       <Spacer height={8} />
-      <BodyRegular>Unfortunately, there's not much to see here.</BodyRegular>
+      <StyledBody>This user doesn't exist yet. If you think they should,<br/>share their collection in a tweet and tag us @usegallery.</StyledBody>
```
Comment (Robinnnnn) [inline src/scenes/NotFound/NotFound.tsx:13]: hyperlink to twitter?

File: src/scenes/NotFound/NotFound.tsx [hunk 2]
```diff
@@ -20,8 +19,13 @@ function NotFound(_: RouteComponentProps) {
   );
 }
 
+const StyledBody = styled(BodyRegular)`
+  white-space: pre-wrap;;
```
Comment (Robinnnnn) [inline src/scenes/NotFound/NotFound.tsx:23]: two semicolons

File: src/scenes/UserGalleryPage/UserGallery.tsx [hunk 3]
```diff
@@ -1,19 +1,30 @@
...
-  gallery: Gallery;
-  isAuthenticatedUsersPage: boolean;
+  username?: string;
 };
 
-function UserGallery({ user, gallery, isAuthenticatedUsersPage }: Props) {
+function UserGallery({ username }: Props) {
+  const user = useUser({ username });
+  const galleries = useGalleries({ userId: user?.id ?? '' }) ?? [];
+  const gallery = galleries[0];
```
Comment (Robinnnnn) [inline src/scenes/UserGalleryPage/UserGallery.tsx:18]: supernit: can combine these lines into `const [gallery] = ...`

File: src/scenes/UserGalleryPage/UserGalleryPageErrorBoundary.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { ApiError } from 'errors/types';
+import { Component } from 'react';
+import NotFound from 'scenes/NotFound/NotFound';
+
+class UserGalleryPageErrorBoundary extends Component {
```
Comment (Robinnnnn) [inline src/scenes/UserGalleryPage/UserGalleryPageErrorBoundary.tsx:5]: ✨

File: src/scenes/NotFound/NotFound.tsx (hunk 2 above)
Comment (kaitoo1) [reply src/scenes/NotFound/NotFound.tsx:23]: ugh the auto linter

File: src/scenes/NotFound/NotFound.tsx (hunk 1 above)
Comment (kaitoo1) [reply src/scenes/NotFound/NotFound.tsx]: good call, added

File: src/scenes/UserGalleryPage/UserGallery.tsx (hunk 3 above)
Comment (kaitoo1) [reply src/scenes/UserGalleryPage/UserGallery.tsx]: nice, done

**Pull request merged**